nifi-nar-bundles/nifi-kafka-bundle/nifi-kafka-0-10-processors/src/main/java/org/apache/nifi/processors/kafka/pubsub/ConsumerLease.java

```diff
@@ -418,82 +419,106 @@ private void writeDemarcatedData(final ProcessSession session, final List<Consum
     }
 
 
-    private void writeRecordData(final ProcessSession session, final List<ConsumerRecord<byte[], byte[]>> records, final TopicPartition topicPartition) {
+    private void writeRecordData(final ProcessSession session, final List<ConsumerRecord<byte[], byte[]>> messages, final TopicPartition topicPartition) {
         // In order to obtain a RecordReader from the RecordReaderFactory, we need to give it a FlowFile.
         // We don't want to create a new FlowFile for each record that we receive, so we will just create
         // a "temporary flowfile" that will be removed in the finally block below and use that to pass to
         // the createRecordReader method.
         final FlowFile tempFlowFile = session.create();
         RecordSetWriter writer = null;
 
+        final BiConsumer<ConsumerRecord<byte[], byte[]>, Exception> handleParseFailure = (consumerRecord, e) -> {
```

Comment:
Unfortunately, I think this PR very much conflicts with NIFI-4046/PR 1906, which I just pushed another commit to, in order to address your feedback. Either PR could be merged first and then the other updated. We both refactored this piece of code to make it re-usable. However, with the new refactoring for 4046, I think I prefer the method there, of having a separate method instead of a lambda because we end up passing in an optional message. Thoughts?